relay-server/src/actors/processor.rs

```diff
@@ -1267,6 +1268,95 @@ impl EnvelopeProcessorService {
         Ok(())
     }
 
+    fn process_replays_combine_items(
+        &self,
+        state: &mut ProcessEnvelopeState,
+    ) -> Result<(), ProcessingError> {
+        let project_state = &state.project_state;
+        let combined_envelope_items =
+            project_state.has_feature(Feature::SessionReplayCombinedEnvelopeItems);
+
+        if combined_envelope_items {
+            // If this flag is enabled, combine both items into a single item,
+            // and remove the original items.
+            // The combined Item's payload is a MsgPack map with the keys
+            // "replay_event" and "replay_recording".
+            // The values are the original payloads of the items.
+            let envelope = &mut state.envelope_mut();
+            if let Some(replay_event_item) =
+                envelope.take_item_by(|item| item.ty() == &ItemType::ReplayEvent)
+            {
```

Comment:
nit: Can also be a guard.

```suggestion
            let Some(replay_event_item) =
                envelope.take_item_by(|item| item.ty() == &ItemType::ReplayEvent) else { return }
```